File: frontend/src/src/actions/game_api.ts
import { gql } from "@apollo/client";

export const FILTER_GAME_API = gql`
  query GetGameApis(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: getGameApis(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      gameApis {
        id
        apiCompanyName
        gameApiName
        gameCompanyName
        gameType
        other
        whetherToUse
        order
        type
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const CREATE_GAME_API = gql`
  mutation CreateGameApi($input: NewGameApiInput!) {
    response: createGameApi(input: $input) {
      id
      apiCompanyName
      gameApiName
      gameCompanyName
      gameType
      other
      whetherToUse
      order
      type
      createdAt
      updatedAt
      deletedAt
    }
  }
`;

export const UPDATE_GAME_API = gql`
  mutation UpdateGameApi($id: ID!, $input: UpdateGameApiInput!) {
    response: updateGameApi(id: $id, input: $input) {
      id
      apiCompanyName
      gameApiName
      gameCompanyName
      gameType
      other
      whetherToUse
      order
      type
      createdAt
      updatedAt
      deletedAt
    }
  }
`; 

File: backend/src/internal/models/game_api.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type GameAPI struct {
	ID              uint            `json:"id" gorm:"primaryKey"`
	UserID          uint            `json:"userId"`
	ApiCompanyName  string          `json:"apiCompanyName"`
	GameApiName     string          `json:"gameApiName"`
	GameCompanyName string          `json:"gameCompanyName"`
	GameType        string          `json:"gameType"`
	Other           string          `json:"other"`
	Type            string          `json:"type"`
	WhetherToUse    bool            `json:"whetherToUse"`
	OrderNum        uint            `json:"orderNum" gorm:"default:1"`
	CreatedAt       time.Time       `json:"createdAt"`
	UpdatedAt       time.Time       `json:"updatedAt"`
	DeletedAt       *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}
